bin/build-lib.sh

```diff
@@ -155,6 +153,18 @@ set_common_vars () {
   fi
 }
 
+# To be called after parsing CLI arguments
+set_derived_vars () {
+  if test -n "$CABAL_WITH_COMPILER"
+  then
+    CABAL_BUILD_OPTIONS+=" --with-compiler $CABAL_WITH_COMPILER"
```

Comment:
`$CABAL_WITH_COMPILER` can be a path. Will this work if the path has spaces? Maybe quote this?

Reply:
We do not usually use paths with spaces, in fact I would not use it with a path. We can fix the issue if and when it arises.